### PR TITLE
Allow discontiguous NestedTensors to empty_like

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
Preivously we disallowd dis-contiguous NTs to passed into to empty_like. This was done out of an abundance of caution, :think:. However it should be safe to create an empty NT for dis-contiguous NTs. Empty like does account for offsets, strides, and sizes in construction of the result and therefore this should be safe.

cc @cpuhrsch @jbschlosser @bhosmer

### CODE REVIEW

File: aten/src/ATen/native/nested/NestedTensorFactories.cpp [hunk 1]
```diff
@@ -51,12 +45,26 @@ Tensor empty_like_nested(
   auto options = verify_empty_parameters(
       self, dtype, layout, device, pin_memory, optional_memory_format);
   auto self_nt = get_nested_tensor_impl(self);
-  Tensor new_buffer = at::empty_like(self_nt->get_buffer(), options);
+  // Since we clone sizes, strides, and offsets it should be safe to use
+  // get_unsafe_storage_as_tensor for the call to empty like.
+  auto memory_format = options.memory_format_opt().value_or(MemoryFormat::Preserve);
+  if (memory_format == MemoryFormat::Contiguous) {
+    Tensor new_buffer = at::empty_like(
```
Comment (drisspg) [inline aten/src/ATen/native/nested/NestedTensorFactories.cpp:52]: This is wrong, this needs to be at::empty( nt.numel, options)

**Review by drisspg — COMMENTED** [review]
Fix the bug when you wake up

**Review by cpuhrsch — APPROVED** [review]

File: aten/src/ATen/native/nested/NestedTensorFactories.cpp [hunk 2]
```diff
@@ -51,12 +45,26 @@ Tensor empty_like_nested(
   auto options = verify_empty_parameters(
       self, dtype, layout, device, pin_memory, optional_memory_format);
   auto self_nt = get_nested_tensor_impl(self);
-  Tensor new_buffer = at::empty_like(self_nt->get_buffer(), options);
+  auto memory_format = options.memory_format_opt().value_or(MemoryFormat::Preserve);
+  if (memory_format == MemoryFormat::Contiguous) {
```
Comment (cpuhrsch) [inline aten/src/ATen/native/nested/NestedTensorFactories.cpp:49]: Maybe for safety add a TORCH_CHECK that `self` is indeed contiguous

**Review by cpuhrsch — APPROVED** [review]

**Pull request closed without merging**